File: JAVA/sistema-bancario/SistemaBancario.java
/**
 *  SISTEMA BANCÁRIO EM JAVA
 * 
 *  Nesse arquivo, fiz um sistema bancário como atividade da DIO. Ele tem uma função básica de todo sistema bancário sem a utilização
 *  de POO (Programação Orientada à Objetos).
 */

import java.util.Scanner; // Importa a classe Scanner para leitura de dados do usuário

public class SistemaBancario {
    // Variáveis estáticas para armazenar as contas
    static int numeroContas = 0; // Contador do número de contas criadas
    static int[] numeros = new int[100]; // Array para armazenar os números das contas
    static String[] titulares = new String[100]; // Array para armazenar os nomes dos titulares das contas
    static double[] saldos = new double[100]; // Array para armazenar os saldos das contas

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in); // Cria um objeto Scanner para leitura de dados do usuário
        int opcao; // Variável para armazenar a opção escolhida pelo usuário

        // Loop principal do menu
        do {
            // Exibe o menu de opções
            System.out.println("\n--- Sistema Bancário ---");
            System.out.println("1. Criar Conta");
            System.out.println("2. Depositar");
            System.out.println("3. Sacar");
            System.out.println("4. Verificar Saldo");
            System.out.println("5. Sair");
            System.out.print("Escolha uma opção: ");
            opcao = scanner.nextInt(); // Lê a opção escolhida pelo usuário

            // Executa a ação correspondente à opção escolhida
            switch (opcao) {
                case 1:
                    criarConta(scanner); // Chama a função para criar uma nova conta
                    break;
                case 2:
                    depositar(scanner); // Chama a função para depositar dinheiro em uma conta
                    break;
                case 3:
                    sacar(scanner); // Chama a função para sacar dinheiro de uma conta
                    break;
                case 4:
                    verificarSaldo(scanner); // Chama a função para verificar o saldo de uma conta
                    break;
                case 5:
                    System.out.println("Saindo do sistema..."); // Mensagem de saída do sistema
                    break;
                default:
                    System.out.println("Opção inválida! Tente novamente."); // Mensagem para opção inválida
            }
        } while (opcao != 5); // Repete o loop até que a opção 5 (sair) seja escolhida

        scanner.close(); // Fecha o objeto Scanner
    }

    // Função para criar uma nova conta
    public static void criarConta(Scanner scanner) {
        if (numeroContas >= 100) { // Verifica se o limite de contas foi atingido
            System.out.println("Limite de contas atingido!");
            return; // Sai da função se o limite foi atingido
        }
        System.out.print("Digite o número da conta: ");
        numeros[numeroContas] = scanner.nextInt(); // Lê o número da nova conta
        scanner.nextLine(); // Consumir a nova linha
        System.out.print("Digite o nome do titular: ");
        titulares[numeroContas] = scanner.nextLine(); // Lê o nome do titular da nova conta
        saldos[numeroContas] = 0.0; // Inicializa o saldo da nova conta como 0.0
        numeroContas++; // Incrementa o contador de contas
        System.out.println("Conta criada com sucesso!"); // Mensagem de sucesso
    }

    // Função para depositar dinheiro em uma conta
    public static void depositar(Scanner scanner) {
        System.out.print("Digite o número da conta: ");
        int numero = scanner.nextInt(); // Lê o número da conta
        int index = encontrarConta(numero); // Encontra o índice da conta no array
        if (index == -1) { // Verifica se a conta não foi encontrada
            System.out.println("Conta não encontrada!");
            return; // Sai da função se a conta não foi encontrada
        }
        System.out.print("Digite o valor do depósito: ");
        double valor = scanner.nextDouble(); // Lê o valor do depósito
        if (valor <= 0) { // Verifica se o valor do depósito é inválido
            System.out.println("Valor inválido!");
            return; // Sai da função se o valor for inválido
        }
        saldos[index] += valor; // Adiciona o valor do depósito ao saldo da conta
        System.out.println("Depósito realizado com sucesso!"); // Mensagem de sucesso
    }

    // Função para sacar dinheiro de uma conta
    public static void sacar(Scanner scanner) {
        System.out.print("Digite o número da conta: ");
        int numero = scanner.nextInt(); // Lê o número da conta
        int index = encontrarConta(numero); // Encontra o índice da conta no array
        if (index == -1) { // Verifica se a conta não foi encontrada
            System.out.println("Conta não encontrada!");
            return; // Sai da função se a conta não foi encontrada
        }
        System.out.print("Digite o valor do saque: ");
        double valor = scanner.nextDouble(); // Lê o valor do saque
        if (valor <= 0 || valor > saldos[index]) { // Verifica se o valor do saque é inválido ou se o saldo é insuficiente
            System.out.println("Valor inválido ou saldo insuficiente!");
            return; // Sai da função se o valor for inválido ou saldo insuficiente
        }
        saldos[index] -= valor; // Subtrai o valor do saque do saldo da conta
        System.out.println("Saque realizado com sucesso!"); // Mensagem de sucesso
    }

    // Função para verificar o saldo de uma conta
    public static void verificarSaldo(Scanner scanner) {
        System.out.print("Digite o número da conta: ");
        int numero = scanner.nextInt(); // Lê o número da conta
        int index = encontrarConta(numero); // Encontra o índice da conta no array
        if (index == -1) { // Verifica se a conta não foi encontrada
            System.out.println("Conta não encontrada!");
            return; // Sai da função se a conta não foi encontrada
        }
        System.out.println("Saldo da conta " + numero + ": " + saldos[index]); // Exibe o saldo da conta
    }

    // Função para encontrar o índice de uma conta no array
    public static int encontrarConta(int numero) {
        for (int i = 0; i < numeroContas; i++) { // Itera sobre todas as contas
            if (numeros[i] == numero) { // Verifica se o número da conta corresponde
                return i; // Retorna o índice da conta
            }
        }
        return -1; // Retorna -1 se a conta não foi encontrada
    }
}

File: PYTHON/sistema-bancario/sistema_bancario.py
menu = '''

[d] Depositar
[s] Sacar
[e] Extrato
[q] Sair

=> '''

saldo = 0  # Inicializa o saldo da conta como zero
limite = 500  # Define o limite de saque como 500
extrato = ''  # Inicializa o extrato vazio
numero_saques = 0  # Inicializa o número de saques realizados como zero
LIMITE_SAQUES = 3  # Define o limite máximo de saques permitidos como 3

# Loop infinito para execução do menu até que o usuário escolha sair
while True:

    opcao = input(menu)  # Exibe o menu e lê a opção escolhida pelo usuário

    if opcao == 'd':  # Se a opção for 'd' (Depositar)
        valor = float(input('Informe o valor do depósito: '))  # Solicita o valor do depósito

        if valor > 0:  # Verifica se o valor do depósito é maior que zero
            saldo += valor  # Adiciona o valor do depósito ao saldo
            extrato += f"Depósito: R$ {valor:.2f}\n"  # Adiciona a operação de depósito ao extrato

        else:
            print("Operação falhou: O valor informado é inválido.")  # Mensagem de erro para valor inválido

    elif opcao == 's':  # Se a opção for 's' (Sacar)
        valor = float(input('Informe o valor do saque: '))  # Solicita o valor do saque

        excedeu_saldo = valor > saldo  # Verifica se o valor do saque excede o saldo disponível
        excedeu_limite = valor > limite  # Verifica se o valor do saque excede o limite permitido
        excedeu_saques = numero_saques >= LIMITE_SAQUES  # Verifica se o número de saques excedeu o limite diário

        if excedeu_saldo:
            print('Operação falhou: Você não tem saldo suficiente.')  # Mensagem de erro para saldo insuficiente

        elif excedeu_limite:
            print('Operação falhou: O valor do saque excede o limite.')  # Mensagem de erro para saque acima do limite

        elif excedeu_saques:
            print('Operação falhou: Números máximos de saques excederam.')  # Mensagem de erro para excesso de saques

        elif valor > 0:  # Verifica se o valor do saque é maior que zero
            saldo -= valor  # Subtrai o valor do saque do saldo
            extrato += f"Saque: R$ {valor:.2f}\n"  # Adiciona a operação de saque ao extrato
            numero_saques += 1  # Incrementa o número de saques realizados

        else:
            print('Operação falhou: O valor informado é inválido.')  # Mensagem de erro para valor inválido

    elif opcao == 'e':  # Se a opção for 'e' (Extrato)
        print("\n=============== EXTRATO ===============")
        # Exibe o extrato se houver operações registradas, caso contrário, informa que não houve movimentações
        print("Não foram realizados movimentações. " if not extrato else extrato)
        print(f"\nSaldo: R$ {saldo:.2f}")  # Exibe o saldo atual da conta
        print("=========================================")

    elif opcao == 'q':  # Se a opção for 'q' (Sair)
        break  # Encerra o loop e sai do programa

    else:
        print('Operação inválida, por favor selecione novamente a operação desejada.')  # Mensagem de erro para opção inválida
